app/presenters/download_options/audio_download_options.rb

```diff
@@ -7,9 +7,34 @@ def initialize(asset)
       super(asset)
     end
 
+    # How many of this asset's parent's members are audio assets.
+    def parent_audio_member_count
+      @audio_members ||= begin
+        members = asset.parent.members.with_representative_derivatives
+        members = members.where(published: true) if current_user.nil?
+        members.count { |m| m.leaf_representative&.content_type&.start_with?("audio/") }
+      end
+    end
+
```

Comment:
I believe this has a problem with "n+1 queries".  This method will do a fetch to the DB. Since there is an AudioDownloadOoptions object instantiated for *each* file in the list, it will be executed once for each file in the list -- even though each query, since based on the parent, will be the same query, it'll be executed multiple redundant times. 

(Correct me if I'm wrong, haven't verified interacting with code, just what it looks like to me). 

Its also not as efficient a query as it could be -- its' fetching ALL memberes, then in-memory filtering to get the count of audio ones. It could, with more difficulty, use postgres jsonb operators to have the DB do the count and get back only the count value. However, I think that wouldn't be good enough, it still wouldn't be good enough, we don't want the n+1. 

This is a problem I've run into before with the "ViewModel" archiecture -- when you need to calculate/fetch something "expensive" that will be the same for each model in a list, but isn't already easily available pre-calculated, you end up with an "n+1" problem. I haven't found any solutions I love yet. 

One is to have an _argument_ passed into the ViewModel for `parent_audio_member_count`. The thing calling it needs to calculate it (once), and pass it in to each AudioDownloadOptions it instantiating. But this kind of complicates and confuses the method signature for instantiating, making it more confusing to re-use the view model in different contexts, which is supposed to be part of the advantage of a view model. And you can end up having to pass this pre-caluclated value through a chain of calls to get from a common ancestor to where it's needed, which is even uglier. 

But since the "X" in "All X Segments as a single file" is already being calculated, maybe available right at the point AudioDownloadOptions is instantiated, perhaps it's feasible here. 

Another solution I"ve played with is passing in a draper-decorated model instead of the raw model. The draper decoration can have a method to calculate/fetch this "expensive" thing and memoize it. you need to make sure each AudioDownloadOptions gets the *same* draper-decorated model, so the memoized value is really reused. 

I think it might also be worth considering if we really need this variant UI for one vs more than one file. While I agree it is a *better* UI you've come up with... is it enough better to be worth this pain? Could we instead come up with a UI where basically the same things are on the page for both one and more-than-one file, so we don't need to know the count inside AudioDownloadOptions at all?

Reply:
We talked about this on Slack, but to summarize the consensus, we went for a simple UI that obviates need to know the count inside a ViewModel which is *itself* inside another ViewModel (!).

Reply:
Using a much simpler solution, on reflection. Full discussion is on Slack. Summary: "Could we instead come up with a UI where basically the same things are on the page for both one and more-than-one file, so we don't need to know the count inside AudioDownloadOptions at all?" We can and we should.